src/mod_roster_rdbms.erl

```diff
@@ -275,40 +246,86 @@ read_subscription_and_groups(LUser, LServer, LJID, GSFunc, GRFunc) ->
 %% Helper functions
 %%==============================================================================
 
-record_to_string(#roster{us = {User, _Server},
-                         jid = JID, name = Name, subscription = Subscription,
-                         ask = Ask, askmessage = AskMessage}) ->
-    Username = mongoose_rdbms:escape_string(User),
-    SJID =
-    mongoose_rdbms:escape_string(jid:to_binary(jid:to_lower(JID))),
-    Nick = mongoose_rdbms:escape_string(Name),
-    SSubscription = mongoose_rdbms:escape_string(case Subscription of
-                        both -> <<"B">>;
-                        to -> <<"T">>;
-                        from -> <<"F">>;
-                        none -> <<"N">>
-                    end),
-    SAsk = mongoose_rdbms:escape_string(case Ask of
-               subscribe -> <<"S">>;
-               unsubscribe -> <<"U">>;
-               both -> <<"B">>;
-               out -> <<"O">>;
-               in -> <<"I">>;
-               none -> <<"N">>
-           end),
-    SAskMessage = mongoose_rdbms:escape_string(AskMessage),
-    [Username, SJID, Nick, SSubscription, SAsk, SAskMessage,
-     mongoose_rdbms:escape_string(<<"N">>),
-     mongoose_rdbms:escape_string(<<"">>),
-     mongoose_rdbms:escape_string(<<"item">>)].
-
-groups_to_string(#roster{us = {User, _Server},
-                         jid = JID, groups = Groups}) ->
-    Username = mongoose_rdbms:escape_string(User),
-    SJID = mongoose_rdbms:escape_string(jid:to_binary(jid:to_lower(JID))),
-    lists:foldl(fun (<<"">>, Acc) -> Acc;
-                    (Group, Acc) ->
-                        G = mongoose_rdbms:escape_string(Group),
-                        [[Username, SJID, G] | Acc]
-                end,
-                [], Groups).
+%% PgSQL returns CHAR as an integer
+%% MySQL returns CHAR as a string
+decode_subscription(<<X>>) -> decode_subscription(X);
```

Comment:
Do not accept binaries and integers in the same function. Previous version was much better.

Reply:
DB returns binaries or integers though...